parsing/parser.mly

```diff
@@ -3431,6 +3441,27 @@ atomic_type:
     | extension
         { Ptyp_extension $1 }
   )
+  { $1 }
+;
+
+atomic_type:
+  | delimited_type 
+      { $1 }
+  | tys = actual_type_parameters
+    tid = mkrhs(type_longident)
+      { mk_ptyp_constr ~loc:$sloc tys tid }
+  | tys = actual_type_parameters
+    HASH
+    cid = mkrhs(clty_longident)
+      { mk_ptyp_class ~loc:$sloc tys cid }
+  | mktyp( /* begin mktyp group */
+      mkrhs(mod_ext_longident) DOT delimited_type 
+        { Ptyp_open ($1, $3) }
```

Comment:
Nitpick: I like menhir's ability to name things instead of using $3 to mean something, and I would encourage you to use it for new parsing rules. (Parts of the current parser were inherited from the previous ocamlyacc grammar which did not support this.)

Reply:
Will do -- I also prefer this style

Reply:
[fd02ca9](https://github.com/ocaml/ocaml/pull/12044/commits/fd02ca917db07e4aa19de0175bcf4f285e2e52ec)